src/main/java/org/jooq/lambda/concurrent/SameExecutorCompletionStage.java

```diff
@@ -0,0 +1,222 @@
+package org.jooq.lambda.concurrent;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.Executor;
+import java.util.function.BiConsumer;
+import java.util.function.BiFunction;
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+final class SameExecutorCompletionStage<T> implements CompletionStage<T> {
+    private final CompletionStage<T> delegate;
+    private final Executor defaultExecutor;
```

Comment:
Should we add logic to make sure it's not null or should it be allowed to be null?